flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/KafkaSourceBuilder.java

```diff
@@ -184,6 +184,18 @@
         return this;
     }
 
+    /**
+     * Set the Kafka subscriber to use to discover new splits.
+     *
+     * @param kafkaSubscriber the {@link KafkaSubscriber} to use for split discovery.
+     * @return this KafkaSourceBuilder.
+     */
+    public KafkaSourceBuilder<OUT> setKafkaSubscriber(KafkaSubscriber kafkaSubscriber) {
+        ensureSubscriberIsNull("subscriber");
```

Comment:
this is probably not needed here?

Reply:
It would be good to keep this check to ensure that the different subscriber setters are mutually exclusive.

Reply:
I'd implement the same behaviour as in `setTopics` and `setTopicPattern`, which use `ensureSubscriberIsNull` too. Maybe instead of `"subscriber"` I'd use `"custom"` or something along these lines.